profiling/src/collections/string_table/mod.rs

```diff
@@ -202,6 +207,78 @@ impl IntoLendingIterator for StringTable {
 mod tests {
     use super::*;
 
+    #[test]
+    fn fuzz_arena_allocator() {
+        bolero::check!()
+            .with_type::<(usize, Vec<String>)>()
+            .for_each(|(size_hint, strings)| {
+                // If the size_hint is insanely large, get allowed allocation
+                // failures.  These are not interesting, so avoid them.
+                if *size_hint > 4 * 1024 * 1024 * 1024 {
+                    return;
+                }
+                let bytes = ChainAllocator::new_in(*size_hint, VirtualAllocator {});
+                let mut allocated_strings = vec![];
+                for string in strings {
+                    let s =
+                        ArenaAllocator::allocate(&bytes, string).expect("allocation to succeed");
+                    assert_eq!(s, string);
+                    allocated_strings.push(s);
+                }
+                assert_eq!(strings.len(), allocated_strings.len());
+                strings
+                    .iter()
+                    .zip(allocated_strings.iter())
+                    .for_each(|(s, t)| assert_eq!(s, t));
+            });
+    }
+
+    /// This is a fuzz test for the allocation optimized `StringTable`.
+    /// It checks both safety (lack of crashes / sanitizer failures),
+    /// as well as functional correctness (the table should behave like an
+    /// ordered set).  
+    /// Limitations:
+    ///   - The length of strings appears to be limited: `assert!(string.len() < 50);` failed, but
+    ///     `100` passed (in 10s)
```

Comment:
I didn't quite understand this comment. Could you elaborate?

Reply:
Bolero, the crate used here to generate `String`s internally has a default range for the length of a string, (0..=64)
https://github.com/camshaft/bolero/blob/f401669697ffcbe7f34cbfd09fd57b93d5df734c/lib/bolero-generator/src/alloc/mod.rs#L17

If a string generated had length >= 50, then `assert!(string.len() < 50);` would have failed but not `100`.

Reply:
Yes, I think that is a better way to explain it, thanks!